db/schema.rb

```diff
@@ -0,0 +1,62 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.1].define(version: 2024_02_24_113652) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "payments", force: :cascade do |t|
+    t.string "name"
+    t.integer "amount"
+    t.bigint "author_id"
+    t.bigint "category_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["author_id"], name: "index_entities_on_author_id"
+    t.index ["category_id"], name: "index_entities_on_category_id"
+  end
+
+  create_table "groups", force: :cascade do |t|
+    t.string "name"
+    t.string "icon"
+    t.bigint "author_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["author_id"], name: "index_groups_on_author_id"
+  end
```

Comment:
- [ ] Good job so far! However, there is a need to make changes to the schema. As per the project requirements, there is a many-to-many relationship between `groups` and `entities`. This implies that a group can contain multiple entities, and one entity can belong to multiple groups. To establish this type of relationship, you need to create an additional table and connect the two schema tables through that new table.
![image](https://github.com/vikas924/budget-app/assets/55840999/13c39050-9ea0-46b9-ba88-7a844e1c286f)


You've already encountered this type of relationship in the "Database" module. I recommend updating your code to reflect this many-to-many relationship. Additionally, after making these changes, thoroughly review and test your entire project to ensure everything works as expected. Keep up the good work! 🚀👍